python/Ganga/new_tests/GPI/newXMLTest/TestSJXMLGenAndLoad.py

```diff
@@ -241,6 +241,9 @@ def test_h_testXMLIndex(self):
             index_cat = raw_j._category
             this_index_cache = (index_cat, index_cls, index_cache)
 
+            print("just-built index: %s" % str(this_index_cache))
+            print("from disk: %s" % str(obj))
```

Comment:
This is happening because the object on the right-hand side is a tuple. Strictly, old-style Python string formatting works with `string % tuple` where the elements of the tuple are inserted into the string. Python is seeing *three* things that you want to inject into the string but only seeing *one* `"%s"`. For full safety with this doing `"%s" % (this_index_cache,)` will ensure the correct behaviour, otherwise using `.format()` is the modern recommended method.

See, for example, http://stackoverflow.com/a/1455623/96704